src/quo2/components/list_items/messages_home_item.cljs

```diff
@@ -0,0 +1,136 @@
+(ns quo2.components.list-items.messages-home-item
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.utils.handlers :refer [<sub]]
+            [status-im.utils.datetime :as time]
+            [quo2.foundations.typography :as typography]
+            [quo2.components.notifications.info-count :refer [info-count]]
+            [quo2.components.icon :as icons]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.avatars.user-avatar :as user-avatar]
+            [quo.react-native :as rn]
+            [status-im.ui.screens.chat.sheets :as sheets]
+            [quo.platform :as platform]))
+
+(def max-subheader-length 100)
+
+(defn truncate-literal [literal]
+  (when literal
+    (let [size (min max-subheader-length (.-length literal))]
+      {:components (.substring literal 0 size)
+       :length     size})))
+
+(defn add-parsed-to-subheader [acc {:keys [type destination literal children]}]
```

Comment:
I understand it's code copied from existing prod code, so I'll just leave a comment without any suggestion for change.

The `add-parsed-to-subheader`, `render-subheader` and `truncate-literal` are begging to be refactored into a pure function outside of the view layer, properly unit tested, and then used in `messages-home-item` namespace to build hiccup. We should be building a dumb view layer as much as possible.

It would be an interesting code challenge :100:

Reply:
Thanks @ilmotta. I will do that.